StashSearch/Plugin.cs

```diff
@@ -62,32 +63,20 @@ private void Start()
             LoadBundle();
         }
 
-        public void Update()
+        public void AttachToInventoryScreen(InventoryScreen inventory)
         {
-            if (Singleton<CommonUI>.Instantiated && !_isActive && !_isInstantiated)
-            {
-                var inventoryScreen = Singleton<CommonUI>.Instance.InventoryScreen;
-                inventoryScreen.GetOrAddComponent<StashComponent>();
-
-                _isActive = true;
-                _isInstantiated = true;
-            }
-
-            if (!_isInstantiated)
-            {
-                return;
-            }
+            // create a new gameobject parented under InventoryScreen with our component on it
+            StashSearchGameObject = new GameObject("StashSearch");
```

Comment:
Technically, we don't need to make a GameObject here to hold the component. It might be nice to parent the actual search GameObjects to these instead of where they are currently, but I didn't try that, since I wasn't sure if the positioning was done relative or not.

Reply:
Nothing wrong here but considering that this method and the `AttachToTraderScreen` method should only run once, you can squeeze the `AddComponent` into the GameObject instantiation instead: `StashSearchGameObject = new GameObject("StashSearch", typeof(StashComponent))`.